packages/core/src/tools/utils.ts

```diff
@@ -304,3 +308,52 @@ export function safeTruncate(candidate: string, length: number) {
   }
   return candidate.slice(0, length)
 }
+
+export function addGlobalEventListeners(
+  events: DOM_EVENT[],
+  listener: (event: Event) => void,
+  { once, capture }: { once?: boolean; capture?: boolean } = {}
+): { stop(): void } {
+  const wrapedListener = monitor(
+    once
+      ? (event: Event) => {
+          stop()
+          listener(event)
+        }
+      : listener
+  )
+
+  const options = supportPassiveEventListeners() ? { capture, passive: true } : capture
+  events.forEach((event) => addEventListener(event, wrapedListener, options))
```

Comment:
should not we make the difference between `window` and `document` addEventListener?

Reply:
I don't think it matters too much but just to be safe, I restored the ability to add listeners on `document`